src/Nethermind/Nethermind.Merge.Plugin/Handlers/NewPayloadHandler.cs

```diff
@@ -164,6 +168,11 @@ public async Task<ResultWrapper<PayloadStatusV1>> HandleAsync(ExecutionPayload r
             return NewPayloadV1Result.Syncing;
         }
 
+        if (_simulateBlockProduction)
+        {
+            _ = CancelSimulation(parentHeader);
+        }
+
```

Comment:
Should we change architecture so we kind of have multiple handlers? Or like an event of NewPayload arrived? Or like CompositeNewPayloadHandler? Or add event here?

Otherwise this is a bit mixing different concepts.

Reply:
Added event